py3status/modules/coin_market.py

```diff
@@ -0,0 +1,168 @@
+
+Examples:
+```
+# view coins in GBP
+coin_market {
+    format_coin = '[\?color=degraded {name}] £{price_gbp:.2f} {percent_change_24h}'
+}
+
+# view coins in GBP and USD
+coin_market {
+    format_coin = '[\?color=degraded {name}] '
+    format_coin += '£{price_gbp:.2f} ${price_usd:.2f} {percent_change_24h}'
+}
+
+# change market name color
+coin_market {
+    color_degraded = '#3DAEE9'
+}
+```
+
+@author x86kernel, lasers
+
+SAMPLE OUTPUT
+[
+    {'color': u'#FFFF00', 'full_text': u'Bitcoin'},
+    {'full_text': u' $2735.77 '},
+    {'color': '#00FF00', 'full_text': u'+2.27%'},
+]
+
+losers
+[
+    {'color': u'#FFFF00', 'full_text': u'Bitcoin'},
+    {'full_text': u' $2701.70 '},
+    {'color': '#FF0000', 'full_text': u'-0.42%'},
+]
+"""
+
+
+class Py3status:
+    cache_timeout = 600
+    format = '{format_coin}'
+    format_coin = '[\?color=degraded {name}] ${price_usd:.2f} {percent_change_24h}'
+    format_separator = ' '
+    markets = ['btc']
+    request_timeout = 5
+
+    def post_config_hook(self):
+        self.first = True
+        self.markets = [x.upper().strip() for x in self.markets]
+        self.url = 'https://api.coinmarketcap.com/v1/ticker/?convert=%s'
+        for item in self.py3.get_placeholders_list(self.format_coin):
+            if (('price' in item and 'price_btc' not in item) or
+                    '24h_volume' in item or 'market_cap' in item) \
+                    and 'usd' not in item:
+                self.url = self.url % (item.split('_')[-1])
+                break
+
+        self.coin_data = self._get_coin_data()
+        limit = 0
+        for market_id in self.markets:
+            index = next((i for (
+                i, d) in enumerate(self.coin_data) if d['symbol'] == market_id), -1)
+            if index > limit:
+                limit = index
+        self.url = self.url + '&limit=%s' % (limit + 1)
+
+    def _get_coin_data(self):
+        try:
+            data = self.py3.request(self.url, timeout=self.request_timeout).json()
+        except self.py3.RequestException:
+            data = {}
+        return data
+
+    def coin_market(self):
+        data = []
+        if not self.first:
+            self.coin_data = self._get_coin_data()
+        else:
+            self.first = False
+
+        if self.coin_data:
+            for market_id in self.markets:
+                for currency in self.coin_data:
+                    if market_id == currency['symbol']:
+                        market = currency
+                        break
+
+                temporary = {}
+                for k, v in market.items():
+                    if 'percent_change' in k and v:
+                        if float(v) >= 0:
+                            _fmt = '\?color=good %s' % v
+                        else:
+                            _fmt = '\?color=bad %s' % v
+                        _fmt = '+' + _fmt if float(v) > 0 else _fmt
+                        temporary[k] = self.py3.safe_format(_fmt + '%')
```

Comment:
Using a threshold might work better.  It would allow the user to have greater control and may well simplify things.

You really should add some comments.